text/0750-feature-flag-advisor.md

```diff
@@ -0,0 +1,343 @@
+
+Adding a new CLI tool also increases the CDK CLI complexity, requiring more maintenance and support for this new feature.
+
+## What alternative solutions did you consider?
+
+Another solution could be to prompt users to configure their feature flags when they create their CDK app using `cdk init`. This
+increases visibility to the feature flags and can be integrated smoothly within their project creation process. However, this does
+not help existing projects and could be considered an inconvenience and unnecessary by users.
+
+## What is the technical solution (design) of this feature?
+
+To implement this feature, we would first generate a Feature Flag Report artifact to be stored in CloudAssembly. The report is a `Record`
+of individual `FeatureFlag` objects that contain the fields `userValue`, `recommendedValue`, and `summary.` When the `cdk flags` command
+is run, the CLI accesses this artifact stored in `manifest.json` and displays the report to the user. A backup of the previous version of
+`cdk.out`will be stored in memory for future comparison. When the user chooses to modify a specific feature flag to our recommended value,
+that value is changed in  `cdk.json` and the entire application resynthesizes. After resynthesizing, the CloudFormation template is
+compared to the one prior to the change and displayed to the user. If the user wishes, they can restore their previous version by
+rejecting the changes.
+
+![Feature Flag Advisor Design](../images/featureflagadvisor.png)
+
+## Additional Considerations
+
+### Increasing visibility to `cdk flags`
+
+Although this feature will increase visibility to feature flags, we still need to bring visibility to the CLI tool itself. Otherwise,
+CDK users continue to remain unaware of feature flags and their potential benefits. To address this, we will display a notice whenever
+`cdk synth` is run. This notice will be discarded once users have configured all of their flags.
+
+```
+> cdk synth
+Notice: You currently have **10** unconfigured feature flags that could potentially impact
+your CDK application. Run `cdk flags` to learn more. 
+
+> cdk flags 
+    Feature Flag                              Recommended                  User
+    @aws-cdk/...                              true                         true
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         - <unset>
+  
+  MENU ----------------------------------------------------------------------
+   >  Set all flags to recommended values
+      Set all unconfigured flags to recommended values
+      Reset all unconfigured flags to defaults (no impact)
+      Modify individual flag
+```
+
+### Ensuring safe configuration changes
```

Comment:
So, are you actually doing this? or is this for the future? If we are - needs more detail.